extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java

```diff
@@ -226,36 +223,100 @@ public void forMethod(ClassCreator classCreator, MethodCreator constructor,
+        putAllHeaderAnnotations(methodLevelHeadersByName, interfaceClass,
+                extractAnnotations(method.annotation(CLIENT_HEADER_PARAMS)));
+
+        headerFillersByName.putAll(methodLevelHeadersByName);
+    }
+
+    /**
+     * create a field in the stub class to contain (interface) java.lang.reflect.Method corresponding to this method
+     * MP Rest Client spec says it has to be in the request context, keeping it in a field we don't have to
+     * initialize it on each call
+     * 
+     * @param classCreator client (or sub-resource client) class creator
+     * @param constructor constructor of the client
+     * @param methodCreator method for which we put the java.lang.reflect.Method to context (aka this method)
+     * @param interfaceClass class of the interface for this client
+     * @param method jandex counterpart of this method
+     * @param invocationBuilder Invocation.Builder in this method
+     * @param methodIndex index of this method
+     */
+    private void addJavaMethodToContext(ClassCreator classCreator, MethodCreator constructor, MethodCreator methodCreator,
+            ClassInfo interfaceClass, MethodInfo method, AssignableResultHandle invocationBuilder, int methodIndex) {
+        FieldDescriptor javaMethodField = createJavaMethodField(classCreator, constructor, interfaceClass, method, methodIndex);
+        ResultHandle javaMethod = methodCreator.readInstanceField(javaMethodField, methodCreator.getThis());
+        ResultHandle javaMethodAsObject = methodCreator.checkCast(javaMethod, Object.class);
+        methodCreator.assign(invocationBuilder,
+                methodCreator.invokeInterfaceMethod(INVOCATION_BUILDER_PROPERTY_METHOD, invocationBuilder,
+                        methodCreator.load(INVOKED_METHOD), javaMethodAsObject));
+    }
+
+    private FieldDescriptor createJavaMethodField(ClassCreator classCreator, MethodCreator constructor,
+            ClassInfo interfaceClass, MethodInfo method, int methodIndex) {
+        ResultHandle interfaceClassHandle = constructor.loadClass(interfaceClass.toString());
+
+        ResultHandle parameterArray = constructor.newArray(Class.class, method.parameters().size());
+        for (int i = 0; i < method.parameters().size(); i++) {
+            String parameterClass = method.parameters().get(i).name().toString();
+            constructor.writeArrayValue(parameterArray, i, constructor.loadClass(parameterClass));
+        }
+
+        ResultHandle javaMethodHandle = constructor.invokeVirtualMethod(
+                MethodDescriptor.ofMethod(Class.class, "getMethod", Method.class, String.class, Class[].class),
+                interfaceClassHandle, constructor.load(method.name()), parameterArray);
+        FieldDescriptor javaMethodField = FieldDescriptor.of(classCreator.getClassName(), "javaMethod" + methodIndex,
+                Method.class);
+        classCreator.getFieldCreator(javaMethodField).setModifiers(Modifier.PRIVATE | Modifier.FINAL);
```

Comment:
Shouldn't this be done in a static field so it is only loaded once?

Reply:
I changed it to generate a `private static final` field and initialize it in a `static` block

Reply:
nope, I need to fix one thing

Reply:
fixed